q2_diversity/_beta/_visualizer.py

```diff
@@ -40,17 +40,25 @@ def bioenv(output_dir: str, distance_matrix: skbio.DistanceMatrix,
     metadata = metadata.filter_columns(column_type='numeric')
     non_numeric_cols = pre_filtered_cols - set(metadata.columns)
 
+    # Drop samples that have any missing values.
+    # TODO use Metadata API if this type of filtering is supported in the
+    # future.
+    # NOTE this whole caste to df to dropnans then caste back to metadata for
```

Comment:
```suggestion
    # NOTE this whole cast to df to dropnans then cast back to metadata for
```
`cast` > `caste`
:heart: Number One Spelling Jerk